flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -119,6 +152,60 @@ class DataStreamOverAggregate(
 
   }
 
+  def createTimeBoundedProcessingTimeOverWindow(inputDS: DataStream[Row]): DataStream[Row] = {
+
+    val overWindow: Group = logicWindow.groups.get(0)
+    val partitionKeys: Array[Int] = overWindow.keys.toArray
+    val namedAggregates: Seq[CalcitePair[AggregateCall, String]] = generateNamedAggregates
+
+    val index = overWindow.lowerBound.getOffset.asInstanceOf[RexInputRef].getIndex
+    val count = input.getRowType().getFieldCount()
+    val lowerboundIndex = index - count
+    
+    
+    val time_boundary = logicWindow.constants.get(lowerboundIndex).getValue2 match {
+      case _: java.math.BigDecimal => logicWindow.constants.get(lowerboundIndex)
+         .getValue2.asInstanceOf[java.math.BigDecimal].longValue()
+      case _ => throw new TableException("OVER Window boundaries must be numeric")
+    }
+
+    val (aggFields, aggregates) =  AggregateUtil.transformToAggregateFunctions(
+         namedAggregates.map(_.getKey),inputType, needRetraction = false)
+         
+         
+    // As we it is not possible to operate neither on sliding count neither
+    // on sliding time we need to manage the eviction of the events that
+    // expire ourselves based on the proctime (system time). 
+    
+    // get the output types
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+
+    val result: DataStream[Row] =
+      if (partitionKeys.nonEmpty) {
+        inputDS.keyBy(partitionKeys:_*)
+          .window(GlobalWindows.create())
+          .trigger(CountTrigger.of(1))
+          .evictor(TimeEvictor.of(Time.milliseconds(time_boundary))) 
```

Comment:
@fhueske i tried also with evicting events and it does work if the .assignTimestampsAndWatermarks is set.
from my point of view if the environment has the ProcessingTime set ...than from my point of view the timestamp should be automatically set to the processing time.
What is your suggestion to handle this issue...we still need to evict based on processing time.

My initial design was to hide this and have this function within the implementation. With that the users do not need to set it and it does not mess up anything ...because in the final end the processing that we build and for which we setup the timestamp is only specific to this query that has the processing time as marker to operate on it.
I would suggest to put it back as it was - what do you think?